Reset input redirection on screen change

phantom10111 opened the pull request:
It's possible for LUA to redirect input and not bring it back when screen is ending (and there are existing bugs like this in Simply Love). Considering that each screen is a separate entity, it seems like this shouldn't be possible and each screen should handle redirection separately. Split redirection state per each screen in the stack and clear it once screen is ending.

teejusb commented on src/ScreenManager.cpp:
```diff
@@ -365,20 +373,15 @@ bool ScreenManager::IsStackedScreen( const Screen *pScreen ) const
 {
-	if(pn >= m_input_redirected.size())
-	{
+	if(g_ScreenStack.empty() || pn >= NUM_PLAYERS)
```
This should also check `pn >= 0`. Ideally we just use the enum directly instead of treating them as numbers but it's fine for now and we can come back to it.

phantom10111 committed:
Reset input redirection on screen change

It's possible for LUA to redirect input and not bring it back when screen is
ending (and there are existing bugs like this in Simply Love). Considering that
each screen is a separate entity, it seems like this shouldn't be possible and
each screen should handle redirection separately. Split redirection state per
each screen in the stack and clear it once screen is ending.
